internal/fdsn/dataselect.go

```diff
@@ -33,9 +33,9 @@ var dataSelectNotSupported = map[string]bool{
 	"minuimumlength": true,
 }
 
-// nslcReg: FDSN spec allows all ascii, but we'll only allow alpha, number, _, ?, *, "," and "--" (exactly 2 hyphens only)
-var nslcReg = regexp.MustCompile(`^([\w*?,]+|--)$`)
-var eventTypeReg = regexp.MustCompile(`^([\w*?, ]+|--)$`) // space allowed
+// nslcReg: FDSN spec allows all ascii, but we'll only allow alpha, number, _,-, ?, *, "," and "--" (exactly 2 hyphens only)
+var nslcReg = regexp.MustCompile(`^([\w*?,]+(?:-[\w*?,]+)*|--)$`)          // space not allowed
+var eventTypeReg = regexp.MustCompile(`^([\w*?, ]+(?:[ -][\w*?,]+)*|--)$`) // space allowed
```

Comment:
Not sure which cases are this change aim to deal with?

I think adding a `\-` to existing regex is sufficient.
`^([\w*?,\- ]+|--)$`

Reply:
This is to allow `-` in the middle, and it needs to be multiple times, yes adding a \- to existing regex is simpler, may be it doesn't matter if `-` is in the start

Reply:
Unless the specification says "no hyphen at the start" (though people won't do it) otherwise we don't need to make sure it's in the middle.